File: Python/5/5_2.py
def process_transactions(transactions):
    for transaction in transactions:
        if transaction['type'] == 'credit':
            if transaction['amount'] > 1000:
                apply_bonus(transaction)
            else:
                apply_standard_processing(transaction)
        elif transaction['type'] == 'debit':
            if transaction['amount'] > 500:
                require_approval(transaction)
            else:
                deduct_amount(transaction)
        else:
            print("Unknown transaction type:", transaction['type'])

File: C#/5/5_2.cs
void ProcessTransactions(List<Transaction> transactions)
{
    foreach (var transaction in transactions)
    {
        if (transaction.Type == "credit")
        {
            CreditPaymant();
        }
        else if (transaction.Type == "debit")
        {
            DebitPaymant();
        }
        else
        {
            Console.WriteLine("Unknown transaction type: " + transaction.Type);
        }
    }
}
void DebitPaymant()
{
    if (transaction.Amount > 500)
    {
        RequireApproval(transaction);
    }
    else
    {
        DeductAmount(transaction);
    }
}
void CreditPaymant()
{
    if (transaction.Amount > 1000)
    {
        ApplyBonus(transaction);
    }
    else
    {
        ApplyStandardProcessing(transaction);
    }
}
